File: js/src/Ice/Property.js
// Copyright (c) ZeroC, Inc. All rights reserved.

export class Property {
    constructor(pattern, usesRegex, defaultValue, deprecated, propertyArray) {
        this.pattern = pattern;
        this.usesRegex = usesRegex;
        this.defaultValue = defaultValue;
        this.deprecated = deprecated;
        this.propertyArray = propertyArray;
    }
}

export class PropertyArray {
    constructor(name, prefixOnly, isOptIn, properties) {
        this.name = name;
        this.prefixOnly = prefixOnly;
        this.isOptIn = isOptIn;
        this.properties = properties;
    }
}

File: csharp/src/Ice/Property.cs
// Copyright (c) ZeroC, Inc.

namespace IceInternal;

public sealed class Property
{
    public Property(string pattern, string defaultValue, bool deprecated, string deprecatedBy)
    {
        _pattern = pattern;
        _defaultValue = defaultValue;
        _deprecated = deprecated;
        _deprecatedBy = deprecatedBy;
    }

    public string
    pattern()
    {
        return _pattern;
    }

    public string
    defaultValue()
    {
        return _defaultValue;
    }

    public bool
    deprecated()
    {
        return _deprecated;
    }

    public string
    deprecatedBy()
    {
        return _deprecatedBy;
    }

    private string _pattern;
    private string _defaultValue;
    private bool _deprecated;
    private string _deprecatedBy;
}
